arm9/src/unk_020040F4.c

```diff
@@ -0,0 +1,1366 @@
+            break;
+        case 5:
+            r5 = FUN_02003D38(27);
+            break;
+        case 6:
+            r5 = FUN_02003D38(28);
+            break;
+    }
+
+    return *r5;
+}
+
+THUMB_FUNC void FUN_020047C8(u8 param0, u8 param1)
+{
+    u8 *ptr;
+    int r4;
+    if (param0 == 1)
+    {
+        ptr = FUN_02003D38(11);
+        r4 = 0;
+    }
+    else
+    {
+        if (param0 != 7)
+        {
+            return;
+        }
+
+        ptr = FUN_02003D38(12);
+        r4 = 7;
+    }
+
+    if (param1 == 0)
+    {
+        FUN_02004110((u16)GF_SndPlayerGetSeqNo(GetSoundPlayer(r4)));
+    }
+
+    NNS_SndPlayerPause(GetSoundPlayer(r4), param1);
+    *ptr = param1;
+}
+
+THUMB_FUNC void FUN_02004810()
+{
+    u8 *ptr = FUN_02003D38(11);
+    u8 *ptr2 = FUN_02003D38(12);
+    *ptr = 0;
+    *ptr2 = 0;
+}
+
+THUMB_FUNC void GF_SndPlayerMoveVolume(int playerNo, u32 param1, u32 param2)
+{
+    NNS_SndPlayerMoveVolume(GetSoundPlayer(playerNo), param1, param2);
+}
+
+THUMB_FUNC void GF_SndPlayerSetInitialVolume(int playerNo, s32 param1)
+{
+    if (param1 < 0)
+    {
+        param1 = 0;
+    }
+
+    if (param1 > 127)
+    {
+        param1 = 127;
+    }
+
+    NNS_SndPlayerSetInitialVolume(GetSoundPlayer(playerNo), param1);
+}
+
+THUMB_FUNC void FUN_02004858(u32 param0, int playerNo)
+{
+    u8 *ret = NNS_SndArcGetSeqParam(param0);
+    u8 r4;
+    if (playerNo == 1 || playerNo == 8)
+    {
+        r4 = 127;
+    }
+    else
+    {
+        if (ret == NULL)
+        {
+            return;
+        }
+
+        r4 = ret[2];
+    }
+
+    if (FUN_020313EC() == 1)
+    {
+        GF_SndPlayerSetInitialVolume(playerNo, r4 / 5);
+    }
+}
+
+THUMB_FUNC void FUN_0200488C(u16 param0, u16 param1)
+{
+    GF_SndPlayerSetInitialVolume((int)FUN_02004018(FUN_020048D0(param0)), param1);
+}
+
+THUMB_FUNC void GF_SndArcPlayerStartSeqEx(int playerNo, u32 param1, u32 param2)
+{
+    NNS_SndArcPlayerStartSeqEx(GetSoundPlayer(playerNo), param1, -1, -1, param2);
+}
+
+THUMB_FUNC u16 GF_SndPlayerCountPlayingSeqByPlayerNo(int playerNo)
+{
+    GF_ASSERT(playerNo >= 0);
+
+    return NNS_SndPlayerCountPlayingSeqByPlayerNo(playerNo);
+}
+
+THUMB_FUNC u8 FUN_020048D0(u16 param0)
+{
+    if (param0 == 0)
+    {
+        return 0xff;
+    }
+
+    u8 *ptr = NNS_SndArcGetSeqParam(param0);
+    if (ptr == NULL)
+    {
+        return 0xff;
+    }
+
+    return ptr[5];
+}
+
+THUMB_FUNC u32 GF_SndPlayerGetSeqNo(u32 *param0)
+{
+    return NNS_SndPlayerGetSeqNo(param0);
+}
+
+THUMB_FUNC u32 FUN_020048F4(u16 param0)
+{
+    return NNS_SndArcGetBankInfo(FUN_02004900(param0));
+}
+
+THUMB_FUNC u16 FUN_02004900(u16 param0)
+{
+    u16 *ptr = NNS_SndArcGetSeqParam(param0);
+    if (ptr == NULL)
+    {
+        return 0;
+    }
+
+    return *ptr;
+}
+
+THUMB_FUNC u32 GF_MIC_StartAutoSampling(u32 param0)
+{
+    return MIC_StartAutoSampling(param0);
+}
+
+THUMB_FUNC u32 GF_MIC_StopAutoSampling(u32 param0)
+{
+#pragma unused(param0)
+    GetSoundDataPointer();
```

Comment:
doesn't make sense for the return value to not be used, did GF really do this? seems a bit weird for them to

Reply:
Yes it's really weird.
The asm looks like this
```
	bl GetSoundDataPointer
	bl MIC_StopAutoSampling
```

So to my knowledge the only way the result of `GetSoundDataPointer` could be used is as an argument of `MIC_StopAutoSampling`. But then `MIC_StopAutoSampling` starts as:
```
	arm_func_start MIC_StopAutoSampling
MIC_StopAutoSampling: ; 0x020D4718
	stmdb sp!, {lr}
	sub sp, sp, #0x4
	ldr r0, _020D4754 ; =MicGetResultCallback
	mov r1, #0x0
	bl MIC_StopAutoSamplingAsync
```
this is ARM and not THUMB so I'm not totally sure but I don't think the value of r0 is used at all. It could also be yet another unused parameter but `MIC_StopAutoSampling` is only called twice so it's hard to infer that.